src/cache/cache.rs

```diff
@@ -111,12 +112,14 @@ impl CacheRead {
     where
         T: Write,
     {
-        let mut file = self
+        let file = self
             .zip
             .by_name(name)
             .chain_err(|| "Failed to read object from cache entry")?;
-        io::copy(&mut file, to)?;
-        Ok(file.unix_mode())
+        let mode = file.unix_mode();
+        let mut decoder = zstd::Decoder::new(file)?;
+        io::copy(&mut decoder, to)?;
```

Comment:
`zstd::stream::copy_decode`?

Maybe check the compression method first?

Reply:
Ah, didn't think about `copy_{en,de}code`, those are reasonable things to use.

I guess we could check the compression method, but what would you want to do?  Throw an error if `!= Stored`?  I think we'll basically do that anyway.  Go ahead and use deflate otherwise?  That seems like it risks silently doing the wrong, slower thing.